app/services/standardize_address_service.rb

```diff
@@ -0,0 +1,52 @@
+require 'net/http'
+require 'uri'
+
+# PLEASE do NOT use this service to standardize addresses in bulk (do NOT use in backfills)
+# as that violates the terms of agreement with USPS and will result in access being revoked.
+class StandardizeAddressService
+  def self.standardize_address_on_intake(intake)
+    build_standardized_address(intake.street_address, intake.city, intake.state, intake.zip_code)
+  end
+
+  private
+
+  def self.build_standardized_address(street_address, city, state, zip_code)
+    usps_address_xml = get_usps_address_xml(street_address, city, state, zip_code)
+    {
+      street_address: usps_address_xml.xpath("//Address/Address2").text,
+      city: usps_address_xml.xpath("//Address/City").text,
+      state: usps_address_xml.xpath("//Address/State").text,
+      zip_code: usps_address_xml.xpath("//Address/Zip5").text,
+      errors: usps_address_xml.xpath("//Error/Description").text
+    }
+  end
+
+  def self.get_usps_address_xml(street_address, city, state, zip_code)
+    usps_userid = "251CODEF8049" # EnvironmentCredentials.dig(:usps, :userid)
+    builder = Nokogiri::XML::Builder.new do |xml|
+      xml.AddressValidateRequest(:USERID => usps_userid) {
+        xml.Revision 1
+        xml.Address(:ID => 0) {
+          xml.Address1
+          xml.Address2 street_address
+          xml.City city
+          xml.State state
+          xml.Zip5 zip_code
+          xml.Zip4
+        }
+      }
+    end
+
+    request_address_xml = Nokogiri::XML(builder.to_xml)
+    usps_request_address = "https://secure.shippingapis.com/ShippingAPI.dll?API=Verify&XML=#{request_address_xml}"
+
+    response = Net::HTTP.get_response(URI(usps_request_address))
+    response_xml = Nokogiri::XML(response.body)
+
+    if response_xml.xpath("//Error").present?
+      Rails.logger.error "Error returned from USPS Address API: #{response_xml.xpath("//Error/Description").text}"
```

Comment:
i think this actually will bypass Sentry and slack because its going to log errors into the rails logger rather than "raise" an error (which would become a Sentry ticket). It would be interesting to have it do both so that we can actively be notified of how often it happens and then deal with it later.